[Huang Nanxi] iP

Nanxi-Huang opened the pull request.

Nanxi-Huang committed:
Implement a skeletal version of Duke that starts by greeting the user, simply echos commands entered by the user, and exits when the user types bye.

Nanxi-Huang committed:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

Nanxi-Huang committed:
Add the ability to mark tasks as done.

Nanxi-Huang committed:
Add support for tracking three types of tasks:
1.ToDos: tasks without any date/time attached to it e.g., visit new theme park
2.Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
3.Events: tasks that start at a specific time and ends at a specific time e.g., team project meeting on 2/10/2019 2-4pm

Nanxi-Huang committed:
Add support for tracking three types of tasks:
1.ToDos: tasks without any date/time attached to it e.g., visit new theme park
2.Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
3.Events: tasks that start at a specific time and ends at a specific time e.g., team project meeting on 2/10/2019 2-4pm

Nanxi-Huang committed:
Teach Duke to deal with errors such as incorrect inputs entered by the user.
e.g., wrong format of commands, invalid command

Nanxi-Huang committed:
Add support for deleting tasks from the list.

Nanxi-Huang committed:
Automated Text UI Testing. increase types of error handled by duke

Nanxi-Huang committed:
Save the tasks in the hard disk

Load last user history from the hard disk when Duke starts up.

Nanxi-Huang committed:
Load past task record from duke.txt

Nanxi-Huang committed:
Parse task records in Task class instead of TaskRecord class

Nanxi-Huang committed:
Merge branch 'branch-Level-7'

Nanxi-Huang committed:
Merge branch 'branch-Level-7.1'

# Conflicts:
#	src/main/java/Task.java

Conflict resolved.

Nanxi-Huang committed:
Teach Duke how to understand dates and times

Nanxi-Huang committed:
Refactor the code to extract out closely related code as classes

Nanxi-Huang committed:
Add JUnit tests to test the behavior of the code

Nanxi-Huang committed:
Improve OOP design

Nanxi-Huang committed:
Add a Parser class to deal with making sense of user commands

Nanxi-Huang committed:
Tweak the code to comply with a coding standard

Nanxi-Huang committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

# Conflicts:
#	src/main/java/Duke.java

Nanxi-Huang committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

# Conflicts:
#	src/main/java/Parser.java

Nanxi-Huang committed:
Add find command for user to search task with keyword

Nanxi-Huang committed:
no message

Nanxi-Huang committed:
Merge branch 'branch-Level-9'

Nanxi-Huang committed:
Complete Week 3 iP levels

Nanxi-Huang committed:
Add methods for user interaction in Ui class

zatkiller commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,95 @@
+            try {
+                int taskIndex = Integer.parseInt(
+                        description.replaceAll("[^0-9]", ""));
+                tasks.Delete(taskIndex);
```
Should Delete method begin with small capital letter?

zatkiller commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,225 @@
+    /**
+     * Prints the number of tasks in the list.
+     */
+    public void Remark() {
```
should Remark begin with a small capital letter as well?

zatkiller reviewed (commented):
Overall code is easy to read with no major code style violation. Good work!

mechastriker3 commented:
Great Job Nanxi!! The code style looks ok in general :)

mechastriker3 commented on src/main/java/Duke.java:
```diff
@@ -17,16 +17,6 @@ public Duke(String filePath) {
 
 
-
-    public void Greet() {
```
Should the method name start with small letters? So would it be better if it was `greet()` instead of `Greet()`

Nanxi-Huang replied:
Thanks for the comment. Will change the method name to start with small letters!

mechastriker3 commented on src/main/java/Duke.java:
```diff
@@ -35,17 +25,6 @@ public void run() throws InvalidCommandException {
     }
 
 
-    public void Exit() {
```
Same for this method, should the method name be `exit()` instead of `Exit()`

Nanxi-Huang replied:
Thanks for pointing out!

mechastriker3 reviewed (commented):
Hi Nanxi, great job!! Generally the code is well written with no major code style violations.

Nanxi-Huang committed:
Automate project builds using Gradle
Use Gradle to automate some of the build tasks of the project

Nanxi-Huang committed:
Use CheckStyle to detect coding style violations

Nanxi-Huang committed:
Edit code style according to CheckStyle

Nanxi-Huang committed:
Improve code style by CheckStyle

Nanxi-Huang committed:
Add a GUI to Duke
Use the JavaFX technology to implement the GUI.

Nanxi-Huang committed:
Improve UI appearance

Nanxi-Huang committed:
Add Junit tests for Todo, Deadline and Event

Nanxi-Huang committed:
Merge branch 'branch-Level-10'

Nanxi-Huang committed:
Add more Junit Test for Duke

Nanxi-Huang committed:
Generate Duke.jar with Gradle

Nanxi-Huang committed:
Use Assertions
Use assert feature (not JUnit assertions) to document important assumptions that should hold at various points in the code.

Nanxi-Huang committed:
Handle more invalid form of todo descriptions

Nanxi-Huang committed:
Merge branch 'master' into branch-A-CodeQuality

Nanxi-Huang committed:
Merge pull request #1 from Nanxi-Huang/branch-A-CodeQuality

Use Assertions

Nanxi-Huang committed:
Improve code quality

Nanxi-Huang committed:
Merge branch 'master' into branch-A-Assertions

Nanxi-Huang committed:
Merge pull request #2 from Nanxi-Huang/branch-A-Assertions

Improve code quality

Nanxi-Huang committed:
no message

Nanxi-Huang committed:
Use lambda expression for some methods

Nanxi-Huang committed:
Set up CI

Nanxi-Huang committed:
A better GUI for duke and add support for managing info about contacts

Nanxi-Huang committed:
Give help to users
Provide in-App guidance to users by adding a command to access a help page.

Nanxi-Huang committed:
Merge branch 'master' into branch-A-Assertions

Nanxi-Huang committed:
Merge pull request #4 from Nanxi-Huang/branch-A-Assertions

no message

Nanxi-Huang committed:
Finalize features and GUI

Nanxi-Huang committed:
Add a User Guide to Duke

Nanxi-Huang committed:
improve User Guide formatting

Nanxi-Huang committed:
Delete javafx-sdk-11.0.2 directory

Nanxi-Huang committed:
update gitignore
